app/screens/home/SettingScreen.tsx

```diff
@@ -149,6 +152,10 @@ const SettingScreen = (props: any): any => {
   const countryId = useAppSelector(
     (state: any) => state.selectedCountry.countryId,
   );
+  const allCountries = useAppSelector(
+    (state: any) =>
+      state.selectedCountry.countries != '' ? JSON.parse(state.selectedCountry.countries) : [],
+  );

```

Comment:
Should we add try/catch to handle parsing errors?

const allCountries = useAppSelector(
 (state: any) => {
  try {
    state.selectedCountry.countries !== '' ? JSON.parse(state.selectedCountry.countries) : [];
  } catch (error) {
    console.error('Failed to parse countries JSON:', error);
    return [];
  }
});